File: apps/ios/Vostok/DesignSystem/Typography.swift
import SwiftUI
import UIKit

/// Typography scale matching the Vostok design tokens (packages/ui-tokens/src/index.ts).
/// Uses the Inter font family bundled with the app.
/// Sizes are absolute points and do **not** scale with Dynamic Type,
/// preserving pixel-accurate layout in message bubbles and tab bars.
enum VostokTypography {
    // MARK: – Token-aligned scale (matches web tokens exactly)

    /// 22px Bold (700) — main section titles, auth headers
    static let heading = interFont(size: 22, weight: .bold)

    /// 16px SemiBold (600) — subsection titles
    static let headline = interFont(size: 16, weight: .semibold)

    /// 15px Medium (≈510) — row titles, list items
    static let title = interFont(size: 15, weight: .medium)

    /// 14px Regular (400) — default body text
    static let body = interFont(size: 14, weight: .regular)

    /// 14px Medium (≈510) — emphasis body text
    static let bodyBold = interFont(size: 14, weight: .medium)

    /// 13px Regular (400) — secondary info, captions
    static let caption = interFont(size: 13, weight: .regular)

    /// 12px Regular (400) — labels, badges
    static let small = interFont(size: 12, weight: .regular)

    /// 11px Medium (≈510) — tiny labels, timestamps
    static let micro = interFont(size: 11, weight: .medium)

    // MARK: – iOS-specific additions (for navigation and system contexts)

    /// 34px Bold — large navigation titles
    static let titleLarge = interFont(size: 34, weight: .bold)

    /// 10px Medium — tab bar labels
    static let tabLabel = interFont(size: 10, weight: .medium)

    // MARK: – Deprecated aliases (map old names to new token names)

    /// Deprecated — use `heading`
    static var titleSection: Font { heading }
    /// Deprecated — use `bodyBold`
    static var bodyEmphasis: Font { bodyBold }
    /// Deprecated — use `bodyBold`
    static var bodyEmphasized: Font { bodyBold }
    /// Deprecated — use `title`
    static var subheadline: Font { title }
    /// Deprecated — use `title`
    static var subheadlineEmphasis: Font { title }
    /// Deprecated — use `title`
    static var subheadlineEmphasized: Font { title }
    /// Deprecated — use `caption`
    static var footnote: Font { caption }

    // MARK: – UIFont equivalents (for UIKit contexts like tab bar appearance)

    static func interUIFont(size: CGFloat, weight: UIFont.Weight) -> UIFont {
        let name: String
        switch weight {
        case .bold:     name = "Inter-Bold"
        case .semibold: name = "Inter-SemiBold"
        case .medium:   name = "Inter-Medium"
        default:        name = "Inter-Regular"
        }
        return UIFont(name: name, size: size) ?? .systemFont(ofSize: size, weight: weight)
    }

    // MARK: – Private

    private static func interFont(size: CGFloat, weight: Font.Weight) -> Font {
        let name: String
        switch weight {
        case .bold:     name = "Inter-Bold"
        case .semibold: name = "Inter-SemiBold"
        case .medium:   name = "Inter-Medium"
        default:        name = "Inter-Regular"
        }
        return .custom(name, fixedSize: size)
    }
}

File: apps/android/app/src/main/java/chat/vostok/android/designsystem/theme/Typography.kt
package chat.vostok.android.designsystem.theme

import androidx.compose.material3.Typography
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import chat.vostok.android.R

/**
 * Vostok typography — Inter font family with token-aligned sizes.
 * Source of truth: packages/ui-tokens/src/index.ts
 */

val InterFontFamily = FontFamily(
    Font(R.font.inter_regular, FontWeight.Normal),
    Font(R.font.inter_medium, FontWeight.Medium),
    Font(R.font.inter_semibold, FontWeight.SemiBold),
    Font(R.font.inter_bold, FontWeight.Bold),
)

/**
 * Named typography tokens matching the design system spec.
 * Use these directly in Compose via `VostokType.heading`, etc.
 */
object VostokType {
    /** 22sp Bold — main section titles */
    val heading = TextStyle(fontFamily = InterFontFamily, fontSize = 22.sp, fontWeight = FontWeight.Bold, lineHeight = 26.4.sp)
    /** 16sp SemiBold — subsection titles */
    val headline = TextStyle(fontFamily = InterFontFamily, fontSize = 16.sp, fontWeight = FontWeight.SemiBold, lineHeight = 22.sp)
    /** 15sp Medium (~510) — row titles, list items */
    val title = TextStyle(fontFamily = InterFontFamily, fontSize = 15.sp, fontWeight = FontWeight.Medium, lineHeight = 18.sp)
    /** 14sp Regular — default body text */
    val body = TextStyle(fontFamily = InterFontFamily, fontSize = 14.sp, fontWeight = FontWeight.Normal, lineHeight = 20.sp)
    /** 14sp Medium (~510) — emphasis body */
    val bodyBold = TextStyle(fontFamily = InterFontFamily, fontSize = 14.sp, fontWeight = FontWeight.Medium, lineHeight = 20.sp)
    /** 13sp Regular — secondary info, captions */
    val caption = TextStyle(fontFamily = InterFontFamily, fontSize = 13.sp, fontWeight = FontWeight.Normal, lineHeight = 18.sp)
    /** 12sp Regular — labels, badges */
    val small = TextStyle(fontFamily = InterFontFamily, fontSize = 12.sp, fontWeight = FontWeight.Normal, lineHeight = 16.sp)
    /** 11sp Medium — tiny labels, timestamps */
    val micro = TextStyle(fontFamily = InterFontFamily, fontSize = 11.sp, fontWeight = FontWeight.Medium, lineHeight = 14.sp)
    /** 10sp Medium — tab labels */
    val tabLabel = TextStyle(fontFamily = InterFontFamily, fontSize = 10.sp, fontWeight = FontWeight.Medium, lineHeight = 12.sp)
}

/**
 * Material 3 Typography mapping for MaterialTheme integration.
 * Maps design tokens to Material 3 type scale slots.
 */
val VostokTypography = Typography(
    headlineLarge = VostokType.heading,
    headlineMedium = VostokType.headline,
    titleLarge = VostokType.heading,
    titleMedium = VostokType.headline,
    titleSmall = VostokType.title,
    bodyLarge = VostokType.body,
    bodyMedium = VostokType.caption,
    bodySmall = VostokType.small,
    labelLarge = VostokType.bodyBold,
    labelMedium = VostokType.caption,
    labelSmall = VostokType.micro,
)
